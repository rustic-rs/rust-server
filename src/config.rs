pub mod auth_config;
pub mod auth_file;
pub mod server_config;
