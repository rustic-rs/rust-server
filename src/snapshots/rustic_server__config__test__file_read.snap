---
source: src/config.rs
expression: config
---
RusticServerConfig {
    server: ConnectionSettings {
        listen: "127.0.0.1:8000",
    },
    storage: StorageSettings {
        data_dir: Some(
            "tests/generated/test_repos/",
        ),
        max_size: None,
    },
    auth: HtpasswdSettings {
        disable_auth: false,
        htpasswd_file: Some(
            "tests/fixtures/test_data/.htpasswd",
        ),
    },
    acl: AclSettings {
        acl_path: Some(
            "tests/fixtures/test_data/acl.toml",
        ),
        private_repo: true,
        append_only: false,
    },
    tls: TlsSettings {
        tls: false,
        tls_key: None,
        tls_cert: None,
    },
    log: LogSettings {
        log_level: None,
        log_file: None,
    },
}
